### PR TITLE
Report a better error when PDB writer is not available

### LINKED ISSUE #10269: Improve native PDB writer exception message when the right interface is not available
We should report a better error than the following, when the expected writer interface is not present.

```
error CS0041: Unexpected error writing debug information -- 'Unable to cast COM object of type 'System.__ComObject' to interface type 'Microsoft.Cci.ISymUnmanagedWriter5'. This operation failed becaus
e the QueryInterface call on the COM component for the interface with IID '{DCF7780D-BDE9-45DF-ACFE-21731A32000C}' failed due to the following error: No such interface supported (Exception from HRESUL
T: 0x80004002 (E_NOINTERFACE)).'
```

### PR DESCRIPTION
**Pull request by tmat**
Fixes #10269

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler 

@MattGertz FYI

File: src/Compilers/Core/Portable/CodeAnalysisResources.resx [hunk 1]
```diff
@@ -453,7 +453,10 @@
     <value>Invalid data at offset {0}: {1}{2}*{3}{4}</value>
   </data>
   <data name="SymWriterNotDeterministic" xml:space="preserve">
-    <value>SymWriter doesn't support deterministic compilation</value>
+    <value>Windows PDB writer doesn't support deterministic compilation (is the compiler host missing Microsoft.DiaSymReader.Native library?)</value>
+  </data>
+  <data name="SymWriterNotAvailable" xml:space="preserve">
+    <value>Windows PDB writer is not available (is the compiler host missing Microsoft.DiaSymReader.Native library?)</value>
```
Comment (agocke) [inline src/Compilers/Core/Portable/CodeAnalysisResources.resx]: Maybe just `Windows PDB writer is not available -- could not find Microsoft.DiaSymReader.Native`?

**Comment by agocke** [comment]
:+1:
Wording seems a bit awkward but we can figure out a better one later.

File: src/Compilers/Core/Portable/CodeAnalysisResources.resx (hunk 1 above)
Comment (MattGertz) [inline src/Compilers/Core/Portable/CodeAnalysisResources.resx]: I'm concerned that "compiler host" isn't a common enough term, but I don't have a better suggestion. We can use it for RC2 and think of something better later, as Andy suggests.

**Comment by MattGertz** [comment]
Approved pending tests. (I can't seem to set labels from this phone.)

**Comment by tmat** [comment]
Changed wording as Andy suggested

**Pull request merged**